command/bitmarkd/main.go

```diff
@@ -290,17 +291,36 @@ func main() {
 		log.Criticalf("zmq.AuthStart: error: %s", err)
 		exitwithstatus.Message("zmq.AuthStart: error: %s", err)
 	}
+
 	if masterConfiguration.DNSPeerOnly {
-		err = p2p.Initialise(&masterConfiguration.Peering, version, masterConfiguration.Fastsync, p2p.DnsOnly)
+		err = p2p.Initialise(&masterConfiguration.Peering, version, p2p.DnsOnly)
 	} else {
-		err = p2p.Initialise(&masterConfiguration.Peering, version, masterConfiguration.Fastsync, p2p.UsePeers)
+		err = p2p.Initialise(&masterConfiguration.Peering, version, p2p.UsePeers)
 	}
 
 	if nil != err {
 		log.Criticalf("p2p initialise error: %s", err)
 		exitwithstatus.Message("p2p initialise error: %s", err)
 	}
 	defer p2p.Finalise()
+
+	for wait := 0; wait < 3; wait++ { // consensus package depended on p2p.Node
+		if nil == p2p.GlobalP2PNode() {
+			log.Warn("p2p node has initialized")
+			wait = 3
+		}
+		time.Sleep(3 * time.Second)
```

Comment:
I don't understand the purpose of `time.Sleep`. Since `p2p.Initialise` is already called, if `p2p.Initialise` didn't return error when called, does it mean `p2p.Initialise` might still have problem even return without error?

Reply:
Will correct it. It should be host. In votingMetrics , in a rare occasion in startup , node.host is nil.